In any cell, the DNA preserves the code used to synthesize proteins. The nucleotide sequence of a protein-encoding gene is transcribed into another nucleotide sequence (RNA transcript or mRNA), which is then used to synthesize a sequence of amino acids to form a protein. This is historically known as the Central Dogma. RNA polymerases transcribe genes.

In eukaryotes, there are three polymerases (I, II, III)

- RNA polymerase I transcribes genes for 18S, 5.8S, and 28S ribosomal RNA; it is located in the nucleus.

- RNA polymerase II transcribes encoding genes for messenger RNA (mRNA) and small RNAs.

- RNA polymerase III synthesizes 5S RNA, and all transfer RNA (tRNA); is located in the nucleoplasm.

The RNA polymerases are aided in the transcription function by a series of proteins called transcription factors, which have several functions mainly to facilitate the binding of RNA polymerase II to the promoter sequence, forming the transcription initiation complex. The initiation complex recognizes the promoter region in the DNA as a signal for the transcription starting point. The consensus sequence TATTAA (called TATA box) is located 25–35 bases upstream of the initiation site, followed by a poly-G/C consensus sequence located 32–38 bases upstream.

The transcribed RNA contains multiple segments called exons and introns. Exons are segments containing the encoding sequence for protein synthesis, while introns are intervening segments of nucleotides that will be excised from the primary RNA transcript (pre-mRNA) in a process called splicing. Splice sequences are found at the 5′ and 3′ tails of introns marked with the dinucleotide GU at its 5′ end and AG at the 3′ end. Alternative splicing is a mechanism used to create different isoforms of the same protein so that several protein isoforms can be created from a limited repertoire of genes. Alternative splicing is now known to deeply influence biological behavior (sex determination, cell differentiation, and programmed cell death).

After a mature mRNA transcript leaves the nucleus, a ribosome attaches to the transcript to begin translation. The initiation site in both prokaryotes and eukaryotes is not composed of a unique initiation sequence but is more probably a multisubstrate enzyme system based on substrate specificity.

A favorable initiation sequence in mammals is known as “Kozak consensus”: 5’ (A/G)CCAUGG 3’. The PIC scans the mRNA in the 5’ region until it reaches the AUG start codon that is complementary with the anticodon of Met-tRNA. The first AUG codon can be skipped (a process called “leaky scanning”) to use a downstream AUG. This creates protein isoforms when the downstream AUG is located in the same open reading frame as the first one, or a completely different protein if it is not in-frame.

Each ribosome is made up of a larger and smaller subunit, which is different for eukaryotes (the 60S and 40S) and prokaryotes (50S and 30S).codon. There is more than one codon encrypting for each amino acid except methionine (Met) and tryptophan (Trp). The three binding sites are the aminoacyl site (A site), peptidyl site (P site), and exit site (E site).

The A site is the first binding site where the initial incoming aminoacyl tRNA pairs with its codon. As the second aminoacyl tRNA binds to the A site, the initial tRNA molecule shifts to the adjacent P site. As the shift happens, the amino acid attached to the tRNA molecule in the P site forms a peptide bond to the amino acid in the A site. This leaves the tRNA at the P site deacylated while tRNA at the A site carries the peptide chain. The deacylated tRNA moves into the E site and gets released as the tRNA in the A site moves over to the P site. The P site holds the tRNAs linked to the growing polypeptide chain. Then a new tRNA molecule attaches to the codon in the unoccupied A site. The codons UAG, UGA, and UAA are used to end the protein translation (termination codons). The process of adding amino acids to the chain is repeated until a stop codon is reached.